demonstrations/tutorial_pulse_programming101.py

```diff
@@ -0,0 +1,362 @@
+r"""Differentiable pulse programming with qubits in PennyLane
+=============================================================
+
+.. meta::
+    :property="og:description": Simulating differentialble pulse programs in PennyLane with qubits
+    :property="og:image": https://pennylane.ai/qml/_images/pauli_shadows.jpg
+
+*Author: Korbinian Kottmann — Posted: 20 February 2023.
+
+In this demo we are going to introduce pulse gates and differentiable pulse programming, showcase
+the current functionality in PennyLane and run the ctrl-VQE algorithm for an example molecule.
+
+Pulses in quantum computers
+---------------------------
+
+In non-measurement-based quantum computers such as superconducting and ion trap systems, qubits are realized through physical systems with a discrete set of energy levels.
+For example, transmon qubits realize an anharmonic oscillator whose ground and first excited states can serve as the two energy
+levels of a qubit. Such a qubit can be controlled via an electromagnetic field tuned to its energy gap. In general, this
+electromangnetic field can be altered in time, leading to a time-dependent Hamiltonian interaction :math:`H(t)`.
+We call driving the system with such an electromagnetic field for a fixed time window a pulse sequence. During a pulse sequence, the state evolves according
+to the time-dependent Schrodinger equation
+
+.. math:: \frac{d}{dt}|\psi\rangle = -i H(t) |\psi\rangle
+
+realizing a unitary evolution :math:`U(t_0, t_1)` from times :math:`t_0` to :math:`t_1` of the input state, i.e. 
+:math:`|\psi(t_1)\rangle = U(t_0, t_1) |\psi(t_0)\rangle`.
+
+In non-measurement-based digital quantum computers, the amplitude and frequencies of predefined pulse sequences are
+fine tuned to realize the native gates of the quantum computer. More specifically, the Hamiltonian interaction :math:`H(t)`
+is tuned such that the respective evolution :math:`U(t_0, t_1)` realizes for example a Pauli or CNOT gate.
```

Comment:
maybe some other word than "realize" ? Maybe
```suggestion
is tuned such that the respective evolution :math:`U(t_0, t_1)` results for example in a Pauli or CNOT gate.
```

Reply:
I think "realize" here makes sense as the pulse sequence realizes some abstract mathematical gate.

Reply:
Fair enough :)